tasks/orchestrator.yml

```diff
@@ -16,3 +16,25 @@
   register: yaml_client_config_secret
   tags:
   - oc
+
+# Setup imagestream
+- name: copy imagestream
+  template:
+    src: osbs-buildroot-imagestream.yml.j2
+    dest: "{{ osbs_openshift_home }}/{{ inventory_hostname }}-{{ osbs_namespace }}-osbs-buildroot-imagestream.yml"
+  environment: "{{ osbs_environment }}"
+  when: osbs_buildroot_repository != ''
+  register: yaml_imagestream
+  tags:
+  - oc
+
+- name: create imagestream
+  command: >
+    oc replace
+    --namespace={{ osbs_namespace }}
+    --force=true
+    --filename={{ osbs_openshift_home }}/{{ inventory_hostname }}-{{ osbs_namespace }}-osbs-buildroot-imagestream.yml
+  environment: "{{ osbs_environment }}"
+  when: yaml_imagestream.changed
```

Comment:
This might need to be `when: osbs_buildroot_repository != '' and yaml_imagestream.changed`, because I don't think the registered variable `yaml_imagestream` will exist if the previous step is not run. (will it?)

Reply:
"If a task fails or is skipped, the variable still is registered with a failure or skipped status, the only way to avoid registering a variable is using tags." 
http://docs.ansible.com/ansible/latest/playbooks_variables.html

so it will exist